### PR TITLE
Bookstore: Reducers & Actions

### PR DESCRIPTION
**Pull request by IndieCoderMM**
# Implement adding and removing using Redux
In this feature branch, the following changes are made.
- Added reducers and actions using the *Ducks* pattern
- Combined reducers and configured store in `redux/configureStore.js`
- Implemented adding and removing books

### CODE REVIEW

File: src/redux/configureStore.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import { combineReducers, configureStore } from '@reduxjs/toolkit';
+import bookReducer from './books/books';
+import categoryReducer from './categories/categories';
+
+const rootReducer = combineReducers({
+  books: bookReducer,
+  categories: categoryReducer,
+});
+
+const configStore = () => configureStore({ reducer: rootReducer });
```
Comment (omar-muhamad) [inline src/redux/configureStore.js]: - [ ] [Optional] In `redux-toolkit` you don't need `combineReducer` you can directly configure your store like that:
```javascript
const store= configureStore({
  books: bookReducer,
  categories: categoryReducer,
});
```

File: src/redux/books/books.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  type: ADD_BOOK,
+  title,
+  author,
+  // eslint-disable-next-line
+  id: currentId++,
+});
+
+export const removeBook = (id) => ({
+  type: REMOVE_BOOK,
+  id,
```
Comment (omar-muhamad) [inline src/redux/books/books.js]: - [ ] [Optional] You created the reducers and action creators using the old way in ordinary `redux`, In `redux-toolkit` you suppose to use `createSlice` to write your reducer.
See this [link](https://redux-toolkit.js.org/api/createSlice) for more info.

File: src/redux/categories/categories.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+const reducer = (state = [], action) => {
+  switch (action.type) {
+    case CHECK_STATUS:
+      return 'Under construction';
+    default:
+      return state;
+  }
+};
+
+export const checkStatus = () => ({ type: CHECK_STATUS });
```
Comment (omar-muhamad) [inline src/redux/categories/categories.js]: - [ ] [Optional] Same here, Please use `createSlice` from the `redux-toolkit` library to write your reducer.

**Review by omar-muhamad — APPROVED** [review]
# Approved ✔
### Hi @IndieCoderMM,

Great job, Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
### Congratulations! 🎉 
### ⚠️Please consider the Optional suggestions below⚠️



## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- You used the correct workflow. ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- N/A

## [Optional] suggestions:

- Please check the inline comments below ⬇.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Comment by IndieCoderMM** [comment]
Thanks for your valuable insights. 😃 I'll implement your suggestions right away.

**Pull request merged**